ambari-server/src/main/python/ambari_server/serverSetup.py

```diff
@@ -1307,7 +1307,7 @@ def check_ambari_java_version_is_valid(java_home, java_bin, min_version, propert
       print 'JDK version found: {0}'.format(actual_jdk_version)
       if actual_jdk_version < min_version:
         print 'Minimum JDK version is {0} for Ambari. Setup JDK again only for Ambari Server.'.format(min_version)
-        properties.process_pair(STACK_JAVA_VERSION, out)
+        properties.process_pair(STACK_JAVA_VERSION, get_java_major_version(out))
```

Comment:
I would prefer storing the value returned by `get_java_major_version` instead of calling it twice.